Breeze.TumbleBit.Client/TumbleBitManager.cs

```diff
@@ -791,6 +792,35 @@ public void ProcessBlock(int height, Block block)
             this.TumblingState.Save();
         }
 
+        public string TumblingDuration(Money walletBalance, Result<ClassicTumblerParameters> tumblerParameters)
+        {
+
+            /* Tumbling cycles occur up to 117 blocks (endBlock) and overlap every 24 blocks (cycleOverlap) :-
+
+                Start block	End block
+                ----------- ---------
+                0	        117
+                24	        141
+                48	        165
+             */
+            const int endBlock = 117;
+            const int cycleOverlap = 24;
+            const double networkFee = 0.0001;
+
+            var demonination = tumblerParameters.Value.Denomination.ToUnit(MoneyUnit.BTC) ;
+            var tumblerFee = tumblerParameters.Value.Fee.ToUnit(MoneyUnit.BTC);
+
+            var  cycleCost = (double)demonination + (double)tumblerFee + networkFee;
+
+            var numberOfCycles = Math.Truncate((double)walletBalance.ToUnit(MoneyUnit.BTC) / cycleCost);
+            var durationInBlocks = endBlock + (numberOfCycles - 1) * cycleOverlap;

```

Comment:
Nothing wrong with the code but this might be more readable
endBlock + ((numberOfCycles - 1) * cycleOverlap);

Reply:
Yes it was like this but Resharper advised me otherwise - I agree with you, let me add back in.